railgun/controllers/configuration/secret_utils.go

```diff
@@ -0,0 +1,166 @@
+package configuration
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/kong/kubernetes-ingress-controller/railgun/controllers"
+	"github.com/kong/kubernetes-ingress-controller/railgun/pkg/configsecret"
+)
+
+// -----------------------------------------------------------------------------
+// Secret Utils - Storage
+// -----------------------------------------------------------------------------
+
+// storeIngressObj reconciles storing the YAML contents of Ingress resources (which are managed by Kong)
+// from multiple versions which remain supported.
+func storeIngressObj(ctx context.Context, c client.Client, log logr.Logger, nsn types.NamespacedName, obj client.Object) (ctrl.Result, error) {
+	// TODO need EVENTS here
+	// TODO need more status updates
+	// TODO: (shane) I want to refactor this into several smaller functions
+	// ^ follow up for these items is in: https://github.com/Kong/kubernetes-ingress-controller/issues/1094
+
+	// if this is an Ingress resource make sure it's managed by Kong
+	if obj.GetObjectKind().GroupVersionKind().Kind == "Ingress" {
+		if !isManaged(obj.GetAnnotations()) {
+			return ctrl.Result{}, nil
+		}
+	}
+
+	// get the configuration secret namespace
+	secretNamespace := os.Getenv(controllers.CtrlNamespaceEnv)
```

Comment:
It doesn't feel right to me that this function operates on namespaced names in two completely different fashions:

ingress namespaced name is a `types.NamespacedName`, while the secret namespaced name is constructed from `os.Getenv` and a const stored elsewhere.

Would it sound appropriate for this function to take two `types.NamespacedName` arguments? One for the (input) ingress being reconciled, and the other for the secret being updated?

Reply:
I may need help to understand the problem better: the secret namespace is explicitly provided via ENV right now, you would prefer this be a function argument is that right?

Reply:
Yes, that summarizes my motivation well.

This preference is based on two assumptions:
- It would allow us to separate the use of the configuration parameter (secret namespace+name) from its presentation (env var + a code constant)
- It would unify presentation of two namespaced names used in this function, which is desirable.

Reply:
Sounds good. I've added this as one of the criteria for the follow up which covers CLI options conformity between v1 and v2:

https://github.com/Kong/kubernetes-ingress-controller/issues/1097

This conversation has been linked therein for context.

Let me know if that seems adequate to move forward or if you would like this handled differently.

Reply:
Let's talk about this in our zoom tomorrow.

Reply:
follow up in: https://github.com/Kong/kubernetes-ingress-controller/pull/1102/files#r597713552